src/lib.rs

```diff
@@ -163,54 +168,61 @@ impl Deref for Client {
 
 /// A Bevy [`Plugin`] for adding support for the Steam SDK.
-pub struct SteamworksPlugin(AppId);
+pub struct SteamworksPlugin {
+    steam: Mutex<Option<(steamworks::Client, SingleClient)>>,
+}
 
 impl SteamworksPlugin {
     /// Creates a new `SteamworksPlugin`. The provided `app_id` should correspond
     /// to the Steam app ID provided by Valve.
-    pub fn new(app_id: impl Into<AppId>) -> Self {
-        Self(app_id.into())
+    pub fn init_app(app_id: impl Into<AppId>) -> Result<Self, SteamAPIInitError> {
+        Ok(Self {
+            steam: Mutex::new(Some(steamworks::Client::init_app(app_id.into())?)),
+        })
+    }
+
+    /// Creates a new `SteamworksPlugin` using the automatically determined app ID.
+    /// If the game isn't being run through steam this can be provided by placing a steam_appid.txt
+    /// with the ID inside in the current working directory.
+    /// Alternatively, you can use `SteamworksPlugin::init_app(<app_id>)` to force a specific app ID.
+    pub fn init() -> Result<Self, SteamAPIInitError> {
+        Ok(Self {
+            steam: Mutex::new(Some(steamworks::Client::init()?)),
+        })
     }
 }
 
 impl Plugin for SteamworksPlugin {
     fn build(&self, app: &mut App) {
-        if app.world.contains_resource::<Client>() {
-            bevy_log::warn!("Attempted to add the Steamworks plugin multiple times!");
-            return;
-        }
-        match steamworks::Client::init_app(self.0) {
-            Err(err) => bevy_log::error!("Failed to initialize Steamworks client: {}", err),
-            Ok((client, single)) => {
-                app.insert_resource(Client(client.clone()))
-                    .insert_resource(register_event_callbacks!(
-                        client,
-                        AuthSessionTicketResponse,
-                        DownloadItemResult,
-                        GameLobbyJoinRequested,
-                        LobbyChatUpdate,
-                        P2PSessionConnectFail,
-                        P2PSessionRequest,
-                        PersonaStateChange,
-                        SteamServerConnectFailure,
-                        SteamServersConnected,
-                        SteamServersDisconnected,
-                        UserAchievementStored,
-                        UserStatsReceived,
-                        UserStatsStored,
-                        ValidateAuthTicketResponse
-                    ))
-                    .insert_non_send_resource(single)
-                    .add_event::<SteamworksEvent>()
-                    .configure_sets(First, SteamworksSystem::RunCallbacks)
-                    .add_systems(
-                        First,
-                        run_steam_callbacks
-                            .in_set(SteamworksSystem::RunCallbacks)
-                            .before(bevy_ecs::event::EventUpdates),
-                    );
-            }
-        }
+        let (client, single) = self.steam.lock().unwrap().take().unwrap();
```

Comment:
I'm not sure why this is necessary. The plugin build occurs on the same thread as everything else.

Likewise, the contract of `Plugin` allows for it to be called multiple times. This will panic if that ever changes.

Reply:
Ah nevermind, I read your earlier comment. That makes sense.

Reply:
We need to move SingleClient and I couldn't figure out a better way. Could you suggest an alternative, please?